erpnext/hr/utils.py

```diff
@@ -6,11 +6,100 @@
 from frappe import _
 from frappe.utils import formatdate, format_datetime
 from frappe.utils import getdate, get_datetime
+from frappe.model.document import Document
+from frappe.desk.form import assign_to
+
+class EmployeeBoardingController(Document):
+	'''
+		Create the project and the task for the boarding process
+		Assign to the concerned person and roles as per the onboarding/separation template
+	'''
+	def validate(self):
+		# remove the task if linked before submitting the form
+		if self.amended_from:
+			for activity in self.activities:
+				activity.task = ''
+
+	def on_submit(self):
+		# create the project for the given employee onboarding
+		project_name = self.doctype + " for " + self.employee_name
```

Comment:
don't use "for" as it is not translatable, use ":" as a separator, also translate the doctype while naming